tools/fregrid/conserve_interp_util.c

```diff
@@ -105,11 +105,58 @@ void read_remap_file( int ntiles_in, int ntiles_out, Grid_config *grid_out,
       free(j_out);
       free(xgrid_area);
       if(opcode & CONSERVE_ORDER2) {
-      	free(xgrid_clon);
-	free(xgrid_clat);
-      }	 
+        free(xgrid_clon);
+        free(xgrid_clat);
+      }
     }//if read from file
   } // ntiles
   if(mpp_pe() == mpp_root_pe())printf("NOTE: Finish reading index and weight for conservative interpolation from file.\n");
 
 }//end read_regrid_weights
+
+
+void malloc_arrays( int nsize, int **i_in, int **j_in, int **i_out, int **j_out,
```

Comment:
Please give a more meaningful name, like malloc_xgrid_arrays.

Reply:
@mlee03 There are lots of messes in the project, and some of them have a natural fix in other languages but not in C. This particular one is one of those.  First we don't have references in C, which would provide part of a natural solution to what you are trying to do. If we did instead of  vars in malloc_arrays() like "int **i_in" we would declare them as "int *& i_in" - a pointer reference so that one could modify the ultimate variable  (like in Fortran) . It would be easier to code the function and to use, and it would be safer. Then if you had templates and variable function args, you can code it like the one I have for C++ (below) - which works for any number and type of arguments (notice sometimes we don't use xgrid_clon and xgrid_clat and we need a second function,  but then function overloading is yet not allowed in C anyways). Its like this:

```
template <typename First>
void reaquire_memory(const size_t sz, First *& first){
  if (first != nullptr)
    free(first);
    first = (First *) malloc(sz * sizeof(First));
}
template <typename First, typename... Rest>
void reaquire_memory(const size_t sz, First*& first, Rest*&... rest){
  reaquire_memory( sz, first );
  reaquire_memory( sz, rest... );
}
```
And I call it like this:
 ```
reaquire_memory(get_MAXXGRID(), i_in, j_in, i_out, j_out, xgrid_area );
```
and like this
```  reaquire_memory(get_MAXXGRID(), i_in, j_in, i_out, j_out, xgrid_area, xgrid_clon, xgrid_clat);```

And while we are on it, the prior developers did not make a good choice by not putting all those fields ( ``` i_in, j_in, i_out, j_out, xgrid_area, xgrid_clon, xgrid_clat ``` ) in a structure, but then since they didn't have good unit tests frameworks (therefore good unit tests) it would be hard to change and guarantee that it works.

In any case, it was very smart of you to notice that this part of the code should be refactored, but you are working in C which has limited capabilities and on a project without a true unit test framework  Perhaps one day you might want to work on the C++ project where Everything is easier :-)

Thanks
MZ